com.unity.ml-agents/Runtime/Sampler.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using Unity.MLAgents;
+using Unity.MLAgents.Inference.Utils;
+using UnityEngine;
+
+namespace Unity.MLAgents
+{
+    /// <summary>
+    /// The types of distributions from which to sample reset parameters.
+    /// </summary>
+    internal enum SamplerType
+    {
+        /// <summary>
+        /// Samples a reset parameter from a uniform distribution.
+        /// </summary>
+        Uniform = 0,
+
+        /// <summary>
+        /// Samples a reset parameter from a Gaussian distribution.
+        /// </summary>
+        Gaussian = 1
+    }
+
+    /// <summary>
+    /// Takes a list of floats that encode a sampling distribution and returns the sampling function.
+    /// </summary>
+    internal sealed class SamplerFactory
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        internal SamplerFactory()
+        {
+        }
+
+        /// <summary>
+        /// Create the sampling distribution described by the encoding.
+        /// </summary>
+        /// <param name="encoding"> List of floats the describe sampling destribution.</param>
+        public Func<float> CreateSampler(IList<float> encoding, int seed)
```

Comment:
I am still not a fan of using a `IList<float>`. If you want to pass an encoding, why not use the InputMessage directly? What if some samplers use ints, I don't like the idea of using floats as encodings if we are going to convert them to ints later.

Reply:
@vincentpierre I addressed this. Let me know if this works for you.